src/main/kotlin/com/openlattice/projector/ProjectEntitySetEntryProcessor.kt

```diff
@@ -0,0 +1,114 @@
+import org.apache.olingo.commons.api.edm.FullQualifiedName
+import org.slf4j.LoggerFactory
+import java.util.*
+
+/**
+ *
+ * @author Drew Bailey &lt;drew@openlattice.com&gt;
+ */
+data class ProjectEntitySetEntryProcessor(
+        val columns: Map<UUID, FullQualifiedName>,
+        val organizationId: UUID,
+        val usersToColumnPermissions: Map<String, List<String>>
+): AbstractRhizomeEntryProcessor<UUID, EntitySet, Void?>(),
+        Offloadable,
+        TransporterDependent<ProjectEntitySetEntryProcessor>
+{
+    @Transient
+    private lateinit var data: TransporterDatastore
+
+    companion object {
+        private val logger = LoggerFactory.getLogger(ProjectEntitySetEntryProcessor::class.java)
+    }
+
+    override fun process(entry: MutableMap.MutableEntry<UUID, EntitySet>): Void? {
+        check(::data.isInitialized) { TransporterDependent.NOT_INITIALIZED }
+        val es = entry.value
+        if ( es.flags.contains( EntitySetFlag.MATERIALIZED )){
+            return null
+        }
+        es.flags.add(EntitySetFlag.MATERIALIZED)
+        entry.setValue(es)
+
+        try {
+            data.linkOrgDbToTransporterDb( organizationId )
+
+            data.datastore().connection.use { conn ->
+                conn.createStatement().use { stmt ->
+                    stmt.executeUpdate(
+                            createEntityTypeView(
+                                    es.name,
+                                    entry.key,
+                                    tableName(es.entityTypeId),
+                                    columns
+                            )
+                    )
+                }
+            }
+
+            data.createOrgDataSource(organizationId).connection.use { conn ->
+                conn.createStatement().use { stmt ->
+                    stmt.executeUpdate(
+                            data.importTablesFromForeignSchema(
+                                    "ol",
+                                    setOf(es.name),
+                                    "transporter",
+                                    data.getOrgFdw( organizationId )
+                            )
+                    )
+                    // TODO - need to apply these as roles due to the maximum row width thing
+//                usersToColumnPermissions.forEach { ( username, allowedCols ) ->
+//                    stmt.addBatch(
+//                            AssemblerConnectionManager.grantSelectSql( es.name, username, allowedCols )
+//                    )
+//                }
+//                stmt.executeBatch()
+                }
+            }
+        } catch ( ex: Exception ) {
+            logger.error("Marking entity set id as not materialized {}", entry.key, ex)
+            es.flags.remove( EntitySetFlag.MATERIALIZED )
+            entry.setValue( es )
+        }
+        return null
+    }
+
+    fun createEntityTypeView(
+            entitySetName: String,
```

Comment:
are data model and/or entity set name changes being handled yet?

Reply:
ish and no
- data model property type additions are handled, thought the view is not refreshed when this happens (so the pt column will be missing) - I can fix the view refreshing
- data model property type removals are *not* handled, didn't realize that actually
- entity set name changes are not handled as I thought they were immutable but can be handled